File: ulv/year_1/1dt901_python/eu222dq_aassignment 2/deck.py
import random
card_values = ["2 ", "3 ", "4 ", "5 ", "6 ", "7 ", "8 ", "9 ", "10 ",
               "jack ", "queen ", "king ", "ace "]
card_colurs = ["of harts ", "of spades ", "of diamonds ", "of clubs "]

deck = []

for value in card_values:
    for colur in card_colurs:
        tmp = value + colur
        deck.append(tmp)

n1 = random.choice(deck)
n2 = random.choice(deck)
n3 = random.choice(deck)
n4 = random.choice(deck)
n5 = random.choice(deck)
print(f"{n1} \n{n2} \n{n3} \n{n4} \n{n5}")

File: soppa/1DT905 - Objektorienerad/a01-labs/lab3/task3/app/src/main/java/com/blackjack/Deck.java
package com.blackjack;


import java.util.ArrayList;
import java.util.Collections;
import java.util.List;



public class Deck {
    private static String[] ranks = {"2", "3", "4", "5", "6", "7", "8", "9", "10", "Jack", "Queen", "King", "Ace"};
    private static String[] suits = {"Hearts", "Diamonds", "Clubs", "Spades"};
    public static List<Card> deck = new ArrayList<>();

    public static void createDeck() {

        for (String suit : suits) {
            for (String rank : ranks) {
                Card card = new Card();
                card.addRank(rank);
                card.addSuit(suit);
                deck.add(card);
            }
        }
    }
    public static void shuffleDeck() {
        Collections.shuffle(deck);
    }


    public static Card getNewCard() {
        Card tempCard = deck.get(0);
        deck.remove(0);
        return tempCard;
    }

    public static Card showNextCard() {
        Card tempCard = deck.get(0);
        return tempCard;
    }


    public static List<Card> twoCardDeal() {
        List<Card> firstDealCards = new ArrayList<>();
        for (int i = 0; i < 3; i++) {
            firstDealCards.add(deck.get(0));
            deck.remove(0);
        }
        return firstDealCards;
    }
}

